### PR TITLE
FIX: Specific email error for replies to digest emails

### PR DESCRIPTION
**Pull request by markvanlan**
Emails the are replies to Digest emails, are getting rejected. Currently They are rejected by `BadDestinationError`. This is the catchall, that multiple cases result in. 

This PR adds a check to make sure sure that the `In-Reply-To` header of incoming messages does not match the `message_id` of a Digest email that was sent out. If this is the case, we can send a much more specific error email, saying that you cannot reply to a digest email.

:tada: Email debugging is **fun**

### CODE REVIEW

**Comment by coding-horror** [comment]
I AM EXCITE 🎉

File: lib/email/receiver.rb [hunk 1]
```diff
@@ -190,6 +191,7 @@ def process_internal
           end
         end
 
+        raise ReplyToDigestError if EmailLog.find_by(email_type: "digest", message_id: @mail.in_reply_to).present?
```
Comment (eviltrout) [inline lib/email/receiver.rb:194]:
```suggestion
        raise ReplyToDigestError if EmailLog.find_by(email_type: "digest", message_id: @mail.in_reply_to).exists?
```

Comment (markvanlan) [reply lib/email/receiver.rb]: @eviltrout I commited this and got `undefined method "exists?" for `EmailLog`. Switching back to `.present?`

Comment (markvanlan) [reply lib/email/receiver.rb]: Ah NVM, I switched the `find_by` to `where` so that `.exists?` should work.

Comment (eviltrout) [reply lib/email/receiver.rb]: Oh my bad it should be `EmailLog.where(email_type: "digest", message_id: @mail.in_reply_to).exists?`

**Pull request merged**